src/objectGuards.ts

```diff
@@ -8,7 +8,7 @@ import {isObject} from './primitiveGuards';
  * Fields in `FROM` that are narrowed in `TO`
  */
 type NarrowedFields<FROM, TO extends FROM, K extends keyof FROM = keyof FROM> = {
-	[P in K]: FROM[P] extends TO[P] ? never : P;
+	[P in K]-?: FROM[P] extends TO[P] ? never : P;
```

Comment:
Review note: this bit of change is more out of paranoia than anything, it is not the crux of the fix.